[SYCL] Change the ext_intel_device_info spec to throw a feature not supported error when a query is not supported (#14576)
intel/llvm: sycl/doc/extensions/supported/sycl_ext_intel_device_info.md

## Error Condition ##

> An exception with a feature not supported error code will be thrown if the device does not support aspect\:\:ext\_intel\_device\_id.

Reviewer: ```suggestion
Throws a synchronous `exception` with the `errc::feature_not_supported` error
code if the device does not have `aspect::ext_intel_device_id`.
```

We use this same type of wording in the core SYCL spec to describe the exceptions that are thrown, so let's use that wording here too.
